pkg/syncer/status/status_process.go

```diff
@@ -44,25 +46,72 @@ func deepEqualFinalizersAndStatus(oldUnstrob, newUnstrob *unstructured.Unstructu
 }
 
-const statusSyncerAgent = "kcp#status-syncer/v0.0.0"
+func (c *Controller) process(ctx context.Context, gvr schema.GroupVersionResource, key string) error {
+	klog.V(3).InfoS("Processing", "gvr", gvr, "key", key)
 
-func NewStatusSyncer(from, to *rest.Config, gvrs []string, kcpClusterName logicalcluster.Name, pclusterID string, advancedSchedulingEnabled bool) (*Controller, error) {
-	from = rest.CopyConfig(from)
-	from.UserAgent = statusSyncerAgent
-	to = rest.CopyConfig(to)
-	to.UserAgent = statusSyncerAgent
+	// from downstream
+	downstreamNamespace, clusterAwareName, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		klog.Errorf("Invalid key: %q: %v", key, err)
+		return nil
+	}
+	downstreamClusterName, name := clusters.SplitClusterAwareKey(clusterAwareName)
+	// TODO(sttts): do not reference the cli plugin here
+	if strings.HasPrefix(workloadcliplugin.SyncerIDPrefix, downstreamNamespace) {
+		// skip syncer namespace
+		return nil
+	}
 
-	fromClient := dynamic.NewForConfigOrDie(from)
-	toClients, err := dynamic.NewClusterForConfig(to)
+	// to upstream
+	nsInformer := c.downstreamInformers.ForResource(schema.GroupVersionResource{Version: "v1", Resource: "namespaces"})
+	nsKey := downstreamNamespace
+	if !downstreamClusterName.Empty() {
+		// If our "physical" cluster is a kcp instance (e.g. for testing purposes), it will return resources
+		// with metadata.clusterName set, which means their keys are cluster-aware, so we need to do the same here.
+		nsKey = clusters.ToClusterAwareKey(downstreamClusterName, nsKey)
+	}
+	nsObj, err := nsInformer.Lister().Get(nsKey)
+	if err != nil {
+		klog.Errorf("Error retrieving namespace %q from downstream lister: %v", nsKey, err)
+		return nil
+	}
+	nsMeta, ok := nsObj.(metav1.Object)
+	if !ok {
+		klog.Errorf("Namespace %q expected to be metav1.Object, got %T", nsKey, nsObj)
+		return nil
+	}
+	namespaceLocator, err := shared.LocatorFromAnnotations(nsMeta.GetAnnotations())
 	if err != nil {
-		return nil, err
+		klog.Errorf(" namespace %q: error decoding annotation: %v", nsKey, err)
+		return nil
+	}
+	if namespaceLocator == nil || namespaceLocator.LogicalCluster != c.upstreamClusterName {
+		// Only sync resources for the configured logical cluster to ensure
+		// that syncers for multiple logical clusters can coexist.
+		return nil
 	}
-	toClient := toClients.Cluster(kcpClusterName)
+	upstreamNamespace := namespaceLocator.Namespace
 
-	// Register the default mutators
-	mutatorsMap := getDefaultMutators(from)
+	// get the downstream object
+	obj, exists, err := c.downstreamInformers.ForResource(gvr).Informer().GetIndexer().GetByKey(key)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		// deleted downstream => remove finalizer upstream
+		klog.InfoS("Downstream GVR %q object %s|%s/%s does not exist. Removing finalizer upstream", gvr.String(), downstreamClusterName, upstreamNamespace, name)
```

Comment:
Don't you want to add this log message inside the conditional block ?

Reply:
👍 , fixed